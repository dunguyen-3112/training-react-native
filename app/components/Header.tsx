import { Image, StyleSheet, Text, View } from 'react-native';
import React, { memo } from 'react';
import CustomText from './CustomText';
import { CustomButton } from '@components';

const Header = () => {
  return (
    <View style={styles.container}>
      <View>
        <CustomText color="black" size={23} weight="800">
          Want to eat
        </CustomText>
        <CustomText color="black" size={23} weight="800">
          healthy Food?
        </CustomText>
      </View>
      <CustomButton
        width={35}
        height={35}
        borderRadius={35 / 2}
        backgroundColor="rgba(0,0,0,0.1)"
      >
        <Image source={require('@assets/icons/question.png')} />
      </CustomButton>
    </View>
  );
};

export default memo(Header);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 20,
  },
});
