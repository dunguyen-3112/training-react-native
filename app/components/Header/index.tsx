import { Image, StyleSheet, View } from 'react-native';
import React, { memo } from 'react';
import { Text } from '@components/common';
import { Button } from '@components/common';

const Header = () => {
  return (
    <View style={styles.container}>
      <Text font={{ fontSize: 23, fontWeight: '800', lineHeight: 31 }}>
        {`Want to eat\nhealthy Food?`}
      </Text>
      <Button
        customStyle={{ width: 35, height: 35 }}
        borderRadius={35 / 2}
        backgroundColor="GRAY"
      >
        <Image source={require('@assets/icons/question.png')} />
      </Button>
    </View>
  );
};

export default memo(Header);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 20,
  },
});
