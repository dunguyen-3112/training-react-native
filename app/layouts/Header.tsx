import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import CustomText from '../components/CustomText';
import { CustomButton } from '@components';

const Header = () => {
  return (
    <View style={styles.container}>
      <View>
        <CustomText text="Want to eat" color="black" size={23} weight="800" />
        <CustomText text="healthy Food?" color="black" size={23} weight="800" />
      </View>
      <CustomButton
        width={35}
        height={35}
        borderRadius={35 / 2}
        backgroundColor="rgba(0,0,0,0.1)"
      >
        <Image source={require('@assets/icons/question.png')} />
      </CustomButton>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 20,
  },
});
