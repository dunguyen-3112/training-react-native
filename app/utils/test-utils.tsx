import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
