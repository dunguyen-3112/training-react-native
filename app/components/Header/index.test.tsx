import React from 'react';
import { render, screen } from '@utils/test-utils';

import Header from './index';

// jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('Test Header', () => {
  test('should', async () => {
    render(<Header />, {
      wrapper: undefined,
    });

    const title = await screen.findByText(`Want to eat\nhealthy Food?`);
    expect(title).toBeDefined();
  });
});
