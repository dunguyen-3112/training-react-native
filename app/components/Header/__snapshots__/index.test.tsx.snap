// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#5C5C5C",
          "color": "#5C5C5C",
          "fontSize": 23,
          "fontWeight": "800",
          "textTransform": "none",
        },
        undefined,
        Object {},
      ]
    }
  >
    Want to eat
healthy Food?
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#ECECEC",
          "borderRadius": 17.5,
          "height": 35,
          "paddingHorizontal": undefined,
          "paddingVertical": undefined,
          "width": 35,
        },
      ]
    }
  >
    <Image
      source={1}
    />
  </View>
</View>
`;
